Source/CombatExtended/CombatExtended/Projectiles/ProjectileCE.cs

```diff
@@ -93,6 +93,34 @@ public Thing intendedTargetThing
             }
         }
 
+        /// <summary>
+        /// Backing field for <see cref="DamageAmount"/>.
+        /// </summary>
+        private int? damageAmount;
+
+        /// <summary>
+        /// Return the damage dealt by this projectile scaled by the quality multiplier of its launcher.
+        /// </summary>
+        public virtual int DamageAmount
```

Comment:
Should return a float.  We use floats internally for damage calculations.  It's a limitation of Core that base weapon damage are integers, but this is an ideal place to make them floats, and will make converting the base damage to floats in the future easier.